nomos-services/network/src/backends/mixnet/mod.rs

```diff
@@ -184,6 +196,62 @@ impl MixnetNetworkBackend {
         }
     }
 
+    async fn stream_send(
+        addr: NodeAddress,
+        data: Box<[u8]>,
+        swarm_commands_tx: &mpsc::Sender<libp2p::Command>,
+        packet_queue: &PacketQueue,
+    ) {
+        let addr = Self::multiaddr_from(addr);
+        let (tx, rx) = oneshot::channel();
+        swarm_commands_tx
+            .send(libp2p::Command::Connect(libp2p::Dial {
+                addr: addr.clone(),
+                retry_count: 3,
+                result_sender: tx,
+            }))
+            .await
+            .unwrap();
```

Comment:
usually better to have `except` with a message for debugging purposes here. Also, would this call return the dial error if it fails? or just the sending channel error?

Reply:
fbbbb03 Thanks. This is just a channel sending error. The actual dialing error is received asynchronously and handled in https://github.com/logos-co/nomos-node/blob/fbbbb039315096a30bd0a3fb8424960a7cbf5a63/nomos-services/network/src/backends/mixnet/mod.rs#L229-L229